### ISSUE TITLE
[SP4] Hibernate doesn't turn off computer

### ISSUE THREAD

**Issue by etnguyen03** [jakeday/linux-surface#320]
When I try to hibernate using

    sudo systemctl hibernate

the Surface (Pro 4) appears to turn off normally, but the SP4 is never turned off; only the display is. The keyboard Fn key and backlight still works.

If I restart using the power button the computer resumes from disk but the mouse, keyboard, and touchscreen freezes and doesn't work.

My `journalctl` logs are [here](https://gist.github.com/etnguyen03/1feedccbc3c430964af45ae134a33fe3).

Has there been a fix for this?

**Comment by tmarkov** [comment]
Does it happen every time? I have observed this happening to be (SB), but only rarely. Usually, holding down the power button will turn off surface for me, and then when I turn it on, it will resume from hibernate.

I have also had a case once where shutdown didn't properly shut it down: keyboard didn't light up, but I kept hearing noises from it until I held down the power button.

**Comment by etnguyen03** [comment]
It appears that it happens every time. While I can hold the power button down for ~10 seconds and hard-restart, and upon restart it attempts to resume from hibernate, but the mouse and keyboard are frozen until I REISUB or hard-restart.

**Comment by thekswenson** [comment]
This happens on my SP4 often, but not always.

**Comment by oleid** [comment]
> the Surface (Pro 4) appears to turn off normally, but the SP4 is never turned off

What's the difference between Surface Pro 4 and SP4? Just wondering...

Does your device turn power off when hibernating using a recent mainline kernel? Also, what distro are you using?

**Comment by etnguyen03** [comment]
> Does your device turn power off when hibernating using a recent mainline kernel?

Yes, turns off and resumes properly using 4.20.0.

> Also, what distro are you using?

Arch.

**Comment by thekswenson** [comment]
I'm trying to test this from scratch so reinstalled Ubuntu 18.10 and ran the install script.

Do I need to do an additional step besides running the install script in order to get hibernate to work?  I forget, but I seem to remember that something like a boot parameter must be set.

**Comment by thekswenson** [comment]
I found it:  i had to tell grub about the swap partition.

**Comment by thekswenson** [comment]
I've reinstalled Ubuntu 18.10, run the install script, and followed the steps to let grub know about the swap partition.

The surface kernel:
- will usually hang while hibernating (holding the power button will then power the machine off), other times will behave normally.
- will resume with wifi working, but no touch or pen input.
- when subsequently shutting down the machine it will again hang until the power button is held for a long period.

The mainline 4.20 kernel works as expected.

**Comment by etnguyen03** [comment]
I managed to figure out how to get the surface to shut down by adding

    [Sleep]
    HibernateMode=shutdown

to `/etc/systemd/sleep.conf.d/hibernatemode.conf`. However now when it resumes from hibernate the touchscreen, mouse, and keyboard are all frozen (except for SysRq apparently). I believe this has been reported; has a fix been found yet?

**Comment by doomdevelop** [comment]
Hi , I have also problem with hibernation .

```
sudo systemctl hibernate
Failed to hibernate system via logind: Not enough swap space for hibernation
```

How can I safely resize swap file ?
My Distro is Ubuntu 18.10.

```
uname -r
4.18.20-surface+

cat /etc/systemd/sleep.conf 
[Sleep]
SuspendState=suspend

 cat /etc/fstab 

UUID=4674d225-be68-49db-a9b6-8a9873d969b9 /               ext4    errors=remount-ro 0       1
# /boot/efi was on /dev/nvme0n1p1 during installation
UUID=2074-D837  /boot/efi       vfat    umask=0077      0       1
/swapfile   
```

**Comment by doomdevelop** [comment]
For any one looking how to resize swap file , [here](https://askubuntu.com/questions/1075505/how-do-i-increase-swapfile-in-ubuntu-18-04) good explanation

**Comment by doomdevelop** [comment]
For any one looking how to resize swap file , [here](https://askubuntu.com/questions/1075505/how-do-i-increase-swapfile-in-ubuntu-18-04) good explanation

**Comment by ge9** [comment]
I'm using SP4 and lubuntu 18.04.
"HibernateMode=shutdown" and "sudo systemctl hibernate" didn't work for me.
"shutdown method = shutdown" (in /etc/uswsusp.conf) and "s2disk" worked (shutdown&resume).
However, after lubuntu resumed, wifi and touchscreen were not working.

**Comment by jakeday** [comment]
This is partially fixed, with the rest being tracked in #270
